### PR TITLE
[WIP] CONSOLE-3392: Implement exact search for users that want more precise result sets

### PR DESCRIPTION
**Pull request by jcaianirh**
Addresses: [CONSOLE-3392](https://issues.redhat.com/browse/CONSOLE-3392)

ezgif-search

### CODE REVIEW

File: frontend/public/components/factory/table-filters.ts [hunk 1]
```diff
@@ -19,8 +19,17 @@ import { alertDescription } from '../monitoring/utils';
...
+export const fuzzyCaseInsensitive = (a: string, b: string): boolean => {
+  // End of quoted string. Compare without leading/trailing quotes
+  if ((a?.startsWith("'") || a?.startsWith('"')) && (a?.endsWith('"') || a?.endsWith("'"))) {
+    return b?.includes(a.substring(1, a.length - 1));
+  }
+  // Beginning of quoted string. Compare without leading quote
+  if (a?.startsWith("'") || a?.startsWith('"')) {
+    return b?.includes(a.substring(1));
+  }
+  return fuzzy(_.toLower(a), _.toLower(b));
```
Comment (TheRealJon) [inline frontend/public/components/factory/table-filters.ts:31]: should we use regex?
```suggestion
  const [exact, token] = a.match(/^['"](.*?)['"]?$/g) ?? [];
  return exact && token ? b?.includes(token) : fuzzy(_.toLower(a), _.toLower(b));
```

File: frontend/public/components/factory/table-filters.ts [hunk 2]
```diff
@@ -19,8 +19,17 @@ import { alertDescription } from '../monitoring/utils';
 import { Target } from '../monitoring/types';
 import { requesterFilter } from '@console/shared/src/components/namespace';
 
-export const fuzzyCaseInsensitive = (a: string, b: string): boolean =>
-  fuzzy(_.toLower(a), _.toLower(b));
+export const fuzzyCaseInsensitive = (a: string, b: string): boolean => {
```
Comment (TheRealJon) [inline frontend/public/components/factory/table-filters.ts:22]: Should we keep this function as-is, and add another filter function for the new behavior?

**Review by TheRealJon — COMMENTED** [review]
This is probably a good start. I was thinking about a few edge cases and what I suspect the current behavior would be. These might have already been hashed out, but just want to make sure:

\# | input | exact filter | fuzzy filter
---|---|---|---|
1 | `"'foo'"` | `'foo'` |
2 | `"""` | `"` | 
3 | `"'` | `'` | |
4 | `" foo "` | ` foo ` (retain whitespace)|
5 | ` "foo"` | | `' "foo"'` 
6 | `"foo" ` | | `"foo" `
7 | `"foo'` | `foo` |
8 | `"foo" bar` | | `"foo" bar`

I think 1-4 are correct, but 5-8 might not be.

File: frontend/public/components/factory/table-filters.ts (hunk 1 above)
Comment (jcaianirh) [reply frontend/public/components/factory/table-filters.ts:31]: @TheRealJon thanks this looks much simpler and more effective

File: frontend/public/components/factory/table-filters.ts (hunk 2 above)
Comment (jhadvig) [reply frontend/public/components/factory/table-filters.ts:22]: I agree, lets keep this one as is and rather add additional filter function.

**Pull request closed without merging**